client.go

```diff
@@ -304,9 +316,60 @@ func (client *client) GetOffset(topic string, partitionID int32, time int64) (in
 	return offset, err
 }
 
+func (client *client) Coordinator(consumerGroup string) (*Broker, error) {
+	client.lock.RLock()
+	coordinator := client.coordinators[consumerGroup]
+	client.lock.RUnlock()
+
+	if coordinator == nil {
+		var err error
+		err = client.RefreshCoordinator(consumerGroup)
+		if err != nil {
+			return nil, err
+		}
+
+		client.lock.RLock()
+		coordinator = client.coordinators[consumerGroup]
+		client.lock.RUnlock()
+	}
+
+	_ = coordinator.Open(client.conf)
+	return coordinator, nil
+}
+
+func (client *client) RefreshCoordinator(consumerGroup string) error {
+	coordinator, err := client.getCoordinator(consumerGroup, client.conf.Metadata.Retry.Max)
+	if err != nil {
+		return err
+	}
+
+	client.lock.Lock()
+	client.coordinators[consumerGroup] = client.registerBroker(coordinator)
```

Comment:
putting this here still feels super-weird to me - why do you prefer this to the `RefreshMetadata` pattern?